### PR TITLE
update msm wrappers

### PR DESCRIPTION
**Pull request by nonam3e**
## Describe the changes

This PR adds not shared bases option in rust bindings

### CODE REVIEW

File: wrappers/rust/icicle-core/src/msm/tests.rs [hunk 1]
```diff
@@ -178,6 +179,172 @@ where
         .unwrap();
 }
 
+pub fn check_msm_batch_shared<C: Curve + MSM<C>>()
+where
+    <C::ScalarField as FieldImpl>::Config: GenerateRandom<C::ScalarField>,
+{
+    // let test_sizes = [1000, 1 << 16]; //TODO - uncomment this line after implementing fast msm
```
Comment (yshekel) [inline wrappers/rust/icicle-core/src/msm/tests.rs:186]: hopefully we merge CPU msm today then let's remove those TODOs and merge this one too

File: wrappers/rust/icicle-core/src/msm/tests.rs [hunk 2]
```diff
@@ -178,6 +179,172 @@ where
...
+        test_utilities::test_set_main_device();
+        cfg.precompute_factor = precompute_factor;
+        let points = generate_random_affine_points_with_zeroes::<C>(test_size, 10);
+        let mut precomputed_points_d =
+            DeviceVec::<Affine<C>>::device_malloc(cfg.precompute_factor as usize * test_size).unwrap();
+        precompute_bases(HostSlice::from_slice(&points), &cfg, &mut precomputed_points_d).unwrap();
+        for batch_size in batch_sizes {
+            let scalars = <C::ScalarField as FieldImpl>::Config::generate_random(test_size * batch_size);
+            // a version of batched msm without using `cfg.points_size`, requires copying bases
+            // let points_cloned: Vec<Affine<C>> = std::iter::repeat(points.clone())
```
Comment (yshekel) [inline wrappers/rust/icicle-core/src/msm/tests.rs:213]: remove if not needed

**Comment by yshekel** [comment]
looks good, let's merge CPU MSM and then merge this too without the TODOs

File: wrappers/rust/icicle-core/src/msm/tests.rs (hunk 2 above)
Comment (ChickenLover) [reply wrappers/rust/icicle-core/src/msm/tests.rs:213]: This was a confusion in previous versions. Safe to remove @nonam3e

**Review by ChickenLover — APPROVED** [review]

**Pull request merged**